### PR TITLE
Fix lambda indent alignment when indent_align_assign is turned on

### PR DESCRIPTION
**Pull request by CharlieYJH**
Fixes the regression mentioned in https://github.com/uncrustify/uncrustify/pull/3136. Prevent manipulation to indenting if indent_align_assign is turned on.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -62,3 +62,4 @@ cmake-build-*/
 
 # Vim
 *.swp
+tags
```
Comment (gmaurel) [inline .gitignore:65]: what are tags?

File: src/indent.cpp [hunk 2]
```diff
@@ -1627,7 +1627,7 @@ void indent_text(void)
             // 1. The matching brace is on the same line as the ending semicolon
             // 2a. If it's an assignment, check that both sides of the assignment operator are on the same line
             // 2b. If it's inside some closure, check that all the frames are on the same line, and it is in the top level closure
-            if (  options::align_assign_span() == 0 && are_chunks_in_same_line(chunk_skip_to_match(frm.top().pc), tail)
+            if (  options::align_assign_span() == 0 && !options::indent_align_assign() && are_chunks_in_same_line(chunk_skip_to_match(frm.top().pc), tail)
```
Comment (gmaurel) [inline src/indent.cpp:1630]: Introduce all the bool operators (such as && here) at the beginning of a new line
Check this for all your proposals.

**Review by gmaurel — CHANGES_REQUESTED** [review]
have a look to the comments

**Comment by CharlieYJH** [comment]
Updated PR with the new format fixes.

File: tests/cpp.test [hunk 3]
```diff
@@ -970,3 +970,4 @@
 60054  interface-keyword-in-cpp.cfg         cpp/interface-keyword-in-cpp.cpp
 60055  issue_3116.cfg                       cpp/issue_3116.cpp
 60056  issue_3116.cfg                       cpp/issue_3116-2.cpp
+60057  issue_3116-2.cfg                     cpp/issue_3116-3.cpp
```
Comment (mwoehlke-kitware) [inline tests/cpp.test:973]: Is there any reason `cpp/issue_3116.cpp` can't be recycled for this purpose?

Comment (CharlieYJH) [reply tests/cpp.test:973]: I suppose my main concern was that `cpp/issue_3116.cpp` also contains cases which weren't mentioned with the config used for this issue, so I don't know what the "expected" format would be. But if you don't deem that to be an issue then yes, we can definitely re-use it.

Comment (mwoehlke-kitware) [reply tests/cpp.test:973]: That feels like a reason *to* combine them, so we can look at the output with additional cases and verify it looks reasonable 🙂.

File: .gitignore (hunk 1 above)
Comment (mwoehlke-kitware) [reply .gitignore]: For the record, possibly related to [ctags](https://en.wikipedia.org/wiki/Ctags) (or at least a similar tool), which is an old mechanism for code browsing (i.e. searching uses of functions, jumping from a function use to its identifier... i.e. the stuff taken for granted in most modern IDEs, except it's been around longer and can be integrated with e.g. vim).

Comment (CharlieYJH) [reply .gitignore]: That's correct. I use ctags for my Vim setup.

File: tests/cpp.test (hunk 3 above)
Comment (CharlieYJH) [reply tests/cpp.test]: Good point, updated the PR with the test replaced by `cpp/issue_3116.cpp`.

**Comment by CharlieYJH** [comment]
Updated PR with new test.

File: tests/expected/cpp/60057-issue_3116.cpp [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+    auto f = [&](int a)
+             {
+                 return b;
+             };
+}
+
+Func(std::count_if(v.begin(), v.end(), [&](const auto &a) {
+                       return a == 3;
+                   }));
```
Comment (mwoehlke-kitware) [inline tests/expected/cpp/60057-issue_3116.cpp:153]: Hmm... this seems a little sketchy, but it's consistent with 0.72, so I guess it's always been this way.

**Review by mwoehlke-kitware — APPROVED** [review]
There is some formatting I find perhaps questionable (but also, perhaps not) in the new expected output, but it's consistent with the pre-#3116 behavior, so no reason to hold this up.

LGTM; thanks for fixing this!

**Comment by gmaurel** [comment]
Please, go to https://github.com/CharlieYJH/uncrustify/actions,
activate the workflow, make a little change and push again to see it is working.
Thanks

**Comment by CharlieYJH** [comment]
Activated the workflow, you can see it here https://github.com/CharlieYJH/uncrustify/actions/runs/925969467.

**Review by gmaurel — APPROVED** [review]

**Pull request merged**